### PR TITLE
Added contract resolve error highlighting + import file fix (alt+enter)

### PR DESCRIPTION
**Pull request by 0v1se**

### CODE REVIEW

File: src/main/kotlin/me/serce/solidity/ide/annotation/annotator.kt [hunk 1]
```diff
@@ -1,21 +1,30 @@
...
 
 class SolidityAnnotator : Annotator {
   override fun annotate(element: PsiElement, holder: AnnotationHolder) {
     if (element is SolElement) {
-      val highlight = highlight(element)
-      if (highlight != null) {
-        val (partToHighlight, color) = highlight
-        holder.createInfoAnnotation(partToHighlight, null).textAttributes = color.textAttributesKey
+      when {
+        element is SolUserDefinedTypeName && element.reference?.resolve() == null -> {
```
[the remark this answers is not in the document]
Comment (0v1se) [reply src/main/kotlin/me/serce/solidity/ide/annotation/annotator.kt]: ok, will check

File: src/main/kotlin/me/serce/solidity/ide/annotation/annotator.kt [hunk 2]
```diff
@@ -1,21 +1,30 @@
...
+          holder.createErrorAnnotation(element, null).let {
```
[the remark this answers is not in the document]
Comment (0v1se) [reply src/main/kotlin/me/serce/solidity/ide/annotation/annotator.kt]: sure, no problem

**Comment by 0v1se** [comment]
I'm concerned about performance. What do you think about resolveTypeNameUsingImports performance and SolidityAnnotator?

It works good for my projects, but don't know if it works ok for bigger projects. May be, some cache should be implemented?

**Comment by SerCeMan** [comment]
@0v1se If you want to make sure that performance doesn't suffer, you can use cachine similar to how it's done in `me.serce.solidity.lang.types.InferenceKt#getType` (using `com.intellij.psi.util.CachedValuesManager`).

**Comment by SerCeMan** [comment]
Hey, @0v1se! Do you need a hand with this?

**Comment by 0v1se** [comment]
@SerCeMan , didn't have any time to finish the work.
Finally, found some time, but didn't finish yet.

Pls, review my new commit. 
Implemented better resolve for structs and enums.
But, there is a problem (failed test shows it). Do you have an idea how to fix it?
The same problem actually applies to structs

**Comment by 0v1se** [comment]
The problem is about SolUserDefinedTypeName can be complex (can be something like A.B from test)
I think I can implement resoleEnum for such SolUserDefinedTypeName. 

But it will be better if A can be resolved to contract A, and B is resolved to enum B from contract A.

What do you think? May be it's overcomplicating for now. This can be done some other time may be

**Comment by SerCeMan** [comment]
> This can be done some other time may be

I agree this is quite a lot for a single commit already. The only problem that I can see is that we don't want to force everyone to use a feature that might not work in some corner cases. What do you think of replacing annotator with a local inspection similar to `me.serce.solidity.ide.inspections.SelfdestructRenameInspection`? That way, users will always be able to opt-out by disabling the inspection.

**Comment by 0v1se** [comment]
I don't know if this functionality can be implemented using inspection. Will check soon and get back

**Comment by SerCeMan** [comment]
> I don't know if this functionality can be implemented using inspection.

I think it should be mostly the same code as it's now, only the code from the annotator will go to a separate inspection class. Ping me if you find any problem with this approach, I'll try to help :)

**Comment by 0v1se** [comment]
@SerCeMan, moved code to separate inspection class.
Also, added some fixes

**Comment by SerCeMan** [comment]
Hey, @0v1se! Thanks for this, looking great! 
I found 1 problem, but it's unrelated to this PR, currently the plugin doesn't resolve `as` imports, so the inspection fails to resolve them.

**Comment by SerCeMan** [comment]
🎉 @0v1se  I'm going to release a new version this weekend, we might sneak https://github.com/intellij-solidity/intellij-solidity/pull/50 in as well, btw feel free to look at this PR and leave comments if you have any

**Pull request merged**